ql/src/test/org/apache/hadoop/hive/ql/io/orc/TestFixAcidKeyIndex.java

```diff
@@ -243,6 +233,12 @@ public void testInvalidKeyIndex() throws Exception {
     checkInvalidKeyIndex(testFilePath);
     // Try fixing, this should result in new fixed file.
     fixInvalidIndex(testFilePath);
+
+    // Multiple stripes
+    createTestAcidFile(testFilePath, 12000, new FaultyKeyIndexBuilder());
+    checkInvalidKeyIndex(testFilePath);
+    // Try fixing, this should result in new fixed file.
+    fixInvalidIndex(testFilePath);
```

Comment:
Shouldn't this check if the index was actually corrected in the new file? It just checks the output of the tool.

Reply:
There is a validation in the tool itself after the recovery, so if the index was still invalid, the output would be different.

https://github.com/asinkovits/hive/blob/HIVE-24475/ql/src/java/org/apache/hadoop/hive/ql/io/orc/FixAcidKeyIndex.java#L265

Reply:
Ah ok, missed that.